src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php

```diff
@@ -96,6 +97,34 @@ public function getDefaultData()
                 'wholesale_price' => 0,
                 'unit_price' => 0,
             ],
+            'shipping' => [
+                'width' => 0,
+                'height' => 0,
+                'depth' => 0,
+                'weight' => 0,
+            ],
+        ];
+    }
+
+    /**
+     * @param ProductForEditing $productForEditing
+     *
+     * @return array<string, mixed>
+     */
+    private function extractShippingData(ProductForEditing $productForEditing): array
+    {
+        $shipping = $productForEditing->getShippingInformation();
+
+        return [
+            'width' => (string) $shipping->getWidth(),
```

Comment:
Shouldn't it be casted into float instead?

Reply:
Is there calculation on it ?

Reply:
nope its decimal numbers in command, so it uses string as input. Unless you mean we should use floats as dimension parameters?